aspnetcore/signalr/diagnostics.md

```diff
@@ -0,0 +1,206 @@
+> A network trace contains the full contents of every message sent by your application. **Never** post raw network traces from production applications to public forums like GitHub.
+
+If you encounter an issue, a network trace can sometimes provide a lot of helpful information. This is particularly useful if you are going to file an issue on our issue tracker.
+
+### Using Fiddler (preferred option)
+
+This method works for all apps.
+
+Fiddler is a very powerful tool for collecting HTTP traces. Install it from [telerik.com/fiddler](https://www.telerik.com/fiddler), launch it, and then run your application and reproduce the issue. Fiddler is available for Windows, and there are beta versions for macOS and Linux.
+
+If you connect using HTTPS, there are some extra steps to ensure Fiddler can decrypt the HTTPS traffic. See the [Fiddler Documentation](https://docs.telerik.com/fiddler/Configure-Fiddler/Tasks/DecryptHTTPS) for more details.
+
+Once you've collected the trace, you can export the trace by choosing "File" > "Save" > "All Sessions..." from the menu bar.
+
+![Exporting all sessions from Fiddler](diagnostics/fiddler-export.png)
+
+### Using tcpdump (macOS and Linux only)
+
+This method works for all apps.
+
+You can collect raw TCP traces using tcpdump by running the following command from a terminal window. You may need to be `root` or prefix the command with `sudo` if you get a permissions error:
+
+```
+tcpdump -i [interface] -w trace.pcap
+```
+
+Replace `[interface]` with the network interface you wish to capture on. Usually this is something like `/dev/eth0` (for your standard Ethernet interface) or `/dev/lo0` (for localhost traffic). See the `tcpdump` man page on your host system for more information.
+
+### Using a browser
+
+This method only works for browser-based apps.
+
+You can also use a browser to capture some network traffic.
+
+> [!NOTE]
+> Most browsers don't include WebSocket and Server-Sent Event messages in exported network traces. If you are using those transports, using a tool like Fiddler or TcpDump (described below) is a better approach.
+
+Most browser Developer Tools have a "Network" tab that allows you to capture network activity between the browser and the server.
+
+#### Microsoft Edge and Internet Explorer
+
+(The instructions are the same for both Edge and Internet Explorer)
+
+1. Press F12 to open the Dev Tools
+2. Click the Network Tab
+3. Refresh the page (if needed) and reproduce the problem
+4. Click the Save icon in the toolbar to export the trace as a "HAR" file:
+
+![The Save Icon on the Microsoft Edge Dev Tools Network Tab](diagnostics/ie-edge-har-export.png)
+
+#### Google Chrome
+
+1. Press F12 to open the Dev Tools
+2. Click the Network Tab
+3. Refresh the page (if needed) and reproduce the problem
+4. Right click anywhere in the list of requests and choose "Save as HAR with content":
+
+!["Save as HAR with Content" option in Google Chrome Dev Tools Network Tab](diagnostics/chrome-har-export.png)
```

Comment:
This image could be shrunk, it seems to be a much larger scale than the others.

Reply:
I'll crop it down. I originally had it this tall to show the full context of the dev tools, but it does look a little out of place.